jsaddle/js/jsaddle-core.js

```diff
@@ -203,7 +203,9 @@ function jsaddle(global, sendRsp, processSyncCommand, RESPONSE_BUFFER_MAX_SIZE)
         args
       ]
     });
-    syncRequests.enqueueArray(newReqs);
+    if (newReqs.length > 0) {
+      syncRequests.enqueueArray(newReqs);
```

Comment:
enqueueArray should be able to handle [] correctly

Reply:
Yeah, that is true, but I see a bug here. I will have to add back the handling of 'Throw' here to avoid mismatches in the syncDepth value in JS and HS state. (I had it earlier https://github.com/obsidiansystems/jsaddle/commit/5fc211e9c0bf58447e74115814d8c63f5995d871)

Reply:
Added handling of Throw before enqueue